syntax = "proto3"; // 声明使用proto3语法


service Crawler{
	rpc request (CrawlerRequest) returns (CrawlerResponse){}
}

message CrawlerRequest{
	string url = 1;
	enum Method {
		GET = 0;
		POST = 1;
	}
	Method method = 2;
	repeated Header headers = 3;
	string data = 4;
	float timeout = 5;
	int32 max_retry = 6;
	bool allow_redirects = 7;
}

message CrawlerResponse{
	enum Status {
		FAILURE = 0;
		SUCCESS = 1;
	}
	Status status = 1;
	string body = 2;
	int32 code = 3;
	repeated Header headers = 4;
	string msg = 5;
	int32 retry_num = 6;
	CrawlerRequest request = 7;
}

message Header {
	string key = 1;
	string value = 2;
}
